File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/entity/EventType.java
/*
 * Copyright (c) Kacper Ziubryniewicz 2020-1-6
 */

package pl.szczodrzynski.edziennik.data.db.entity;

import android.graphics.Color;

import androidx.room.ColumnInfo;
import androidx.room.Entity;

@Entity(tableName = "eventTypes",
        primaryKeys = {"profileId", "eventType"})
public class EventType {
    public int profileId;

    @ColumnInfo(name = "eventType")
    public long id;

    @ColumnInfo(name = "eventTypeName")
    public String name;
    @ColumnInfo(name = "eventTypeColor")
    public int color;

    public EventType(int profileId, long id, String name, int color) {
        this.profileId = profileId;
        this.id = id;
        this.name = name;
        this.color = color;
    }

    public EventType(int profileId, int id, String name, String color) {
        this(profileId, id, name, Color.parseColor(color));
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/entity/EventType.kt
/*
 * Copyright (c) Kuba Szczodrzyński 2021-4-19.
 */
package pl.szczodrzynski.edziennik.data.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity(
    tableName = "eventTypes",
    primaryKeys = ["profileId", "eventType"]
)
class EventType(
    val profileId: Int,

    @ColumnInfo(name = "eventType")
    val id: Long,

    @ColumnInfo(name = "eventTypeName")
    val name: String,
    @ColumnInfo(name = "eventTypeColor")
    val color: Int,
    @ColumnInfo(name = "eventTypeOrder")
    var order: Int = id.toInt(),
    @ColumnInfo(name = "eventTypeSource")
    val source: Int = SOURCE_REGISTER
) {
    companion object {
        const val SOURCE_DEFAULT = 0
        const val SOURCE_REGISTER = 1
        const val SOURCE_CUSTOM = 2
        const val SOURCE_SHARED = 3
    }
}
